### ISSUE TITLE
Stuck on "Loading..." screen after update (Windows 10)

### ISSUE THREAD

**Issue by Deozaan** [keybase/client#5917]
Last week I updated Keybase and ran it for the first time in several months. It got stuck on the Loading screen. I was able to fix it by uninstalling, moving the Keybase folder from AppData/Roaming to AppData/Local, then reinstalling Keybase.

Today Keybase updated itself and now it's stuck on the Loading screen again. And the K drive has disappeared. And the little popup window when I click the icon in the tray says I'm not logged in. I've also noticed that there is a Keybase folder in my Roaming directory again. But there's also one in my Local directory. They have significantly different contents.

First of all, I'm not sure uninstalling, moving directories, and reinstalling would resolve the issue this time because I was under the impression that that only needed to be done because of a change in where Keybase stored information from several months ago. It wouldn't have changed again since last week, would it?

Secondly, I'm not going to keep uninstalling and moving directories and reinstalling every time there's an update.

Thirdly, I did a search and I do see a few other issues open about getting stuck on the loading screen. But I can't tell if they're related or not or they seem to be OS specific for MacOS or Linux.

my log id: 69ff1d5d1990ca20430e091c

**Comment by maxtaco** [comment]
cc: @zanderz  seems like a failed key repair

**Comment by zanderz** [comment]
@Deozaan  I see this in the service log, which I haven't seen before and seems strange:
```
2017-02-17T12:32:59.391782-07:00 - [DEBU keybase util.go:202] 024 - Wrote to C:\Users\Deozaan\AppData\Local\Keybase\keybased.info -> ERROR: rename C:\Users\Deozaan\AppData\Local\Keybase\keybased.info.UWVH56CR5M4CYX6QHZPJKF5QDUU2NCA6 C:\Users\Deozaan\AppData\Local\Keybase\keybased.info: The create operation failed because the name contained at least one mount point which resolves to a volume to which the specified device object is not attached.
```

Are you running as a user in the admin group? Was the installer run under the same user account?

Is this a laptop which went to sleep or had network access interrupted?

**Comment by Deozaan** [comment]
@zanderz Yes, my user account is an Administrator account. And yes, the installer was run under the same account. It is not a laptop.

My guess is that this has something to do with me using a junction to keep files off of my C drive which is a relatively small SSD. I use a junction on virtually every directory in my AppData folder that Windows will let me move.

**Comment by zanderz** [comment]
is there a way to try moving Keybase back to a non-junction location? I think your guess is probably a good one.

**Comment by Deozaan** [comment]
I tried moving the junctioned folders (found in both Local and Roaming subdirectories of AppData) back to their default locations and Keybase was still stuck on the Loading screen. So then I uninstalled Keybase and the Dokan Library then rebooted and reinstalled Keybase and now it's stuck on the Loading screen again.

Here's my latest log id: b32f770e33d2f6b916bebc1c

Any other suggestions or troubleshooting steps I should take?

**Comment by zanderz** [comment]
Your log at the end is filled with attempts to migrate things from `C:\Users\Deozaan\AppData\Roaming\Keybase` to `C:\Users\Deozaan\AppData\Local\Keybase`. Is there a chance you have been reverting to a much older version, which still writes to %APPDATA%? I understand if you don't want to keep moving files manually and reinstalling, but it should only be necessary one time. To make sure, I suggest you uninstall once again, delete `C:\Users\Deozaan\AppData\Roaming\Keybase` altogether (if your .mpack and .ss files have been moved already), and install the latest version.

**Comment by Deozaan** [comment]
I initially encountered this problem (getting stuck on the loading screen) about a month ago when I fired up Keybase again for the first time in several months. I learned about having to migrate from Roaming to Local, so I uninstalled everything, moved the Keybase folder from Roaming to Local (or rather, moved the junction), and installed Keybase again using the then-latest version of Keybase, and everything worked.

About two weeks later, there was a popup telling me about an update, which I told it to go ahead and install. At that point I got stuck on the loading screen again and have been stuck there ever since, even after yet another update.

I was under the impression that the latest version as of about one month ago wouldn't write to the Roaming directory anymore. But it's also possible that I didn't properly clean up the Keybase folder from Roaming. I certainly was surprised after "fixing" it last month to see another folder pop up in the Roaming directory again.

Anyway, the .ss file was moved already but the .mpack file hadn't been. I've copied it over and deleted the Keybase folder from my Roaming directory, and I've uninstalled Dokan and Keybase. I'm about to reboot and then I'll install a freshly downloaded copy of the Keybase installer and report back.

**Comment by zanderz** [comment]
To be clear, some files are still written to %APPDATA%\Keybase by the GUI, but they should not trigger the migration like an .mpack will. Sorry for all the troubles, it sounds like it should work after that.

**Comment by Deozaan** [comment]
Still stuck on the loading screen.

Log ID: f2ef5ecfcc37ecde5d8abe1c

Is there a way I can back up my secret key(s), completely eradicate all traces of Keybase from my system (including what's in the Local folder), and try a totally fresh install of Keybase?

**Comment by zanderz** [comment]
Yeah, this log just continues repeating the same thing as the previous one:
` loading config file: C:\Users\Deozaan\AppData\Local\Keybase\config.json` every 10 seconds. It doesn't seem to be doing the migration any longer.

You could try backing up the .mpack and .ss files elsewhere and doing the cleanup you mention, but there may be something else happening on your machine, it is unclear why it wouldn't be able to write to %LOCALAPPDATA%\Keybase. Do you see config.json and other files in there?

**Comment by Deozaan** [comment]
I do see config.json and other files in there. over 53k files. And a few more being added every few seconds. Is that normal?

![keybase files gone wild](https://cloud.githubusercontent.com/assets/227734/23532710/200ac16c-ff6a-11e6-88df-c3566998afd9.png)

**Comment by zanderz** [comment]
whoops, no. Those are temporary files that are supposed to be renamed to `keybased.info` after writing succeeds. I would say to delete all those after you finish uninstalling. And/or maybe deleting keybased.info would help? Never seen this.

**Comment by Deozaan** [comment]
The weird thing about all those temp files being created is that Keybase shouldn't even be running right now. I closed it because it was just stuck on the loading screen. So how is it writing files every few seconds?

Also, I don't see any `keybased.info` file. Just a bunch of `keybased.info.HASH` files.

I created a 0 byte file named `keybase.info` and after waiting a bit, it's still 0 bytes and more temp files are still being created every few seconds.

So I guess I'm going to try uninstalling one more time, and completely wiping out the Keybase folders with the exception of backing up my .ss and .mpack files.

**Comment by zanderz** [comment]
Are you sure you are no longer using junctions or symlinks for this path?
It may be worth looking in the system event log for clues/errors.

**Comment by Deozaan** [comment]
> Are you sure you are no longer using junctions or symlinks for this path?

Pretty sure, especially now that I completely deleted the folders this time. When I reinstalled it, it's getting stuck on the Loading screen again. The only difference this time was that it popped open my browser to the Getting Started page, presumably because the started.txt didn't exist in the Local Keybase folder. But the temp files are already coming back with a vengeance. Still no `keybased.info` in the directory.

I tried manually renaming one of the temp files to `keybased.info` and still no change.

> It may be worth looking in the system event log for clues/errors.

I perused through my event log but didn't see anything that appeared to be related. But I can't say for sure that I know what I'm looking for, or if I even looked in the right place.

I tried that command you suggested in the Keybase shell. It didn't work. I got an error:

> C:\Users\Deozaan\AppData\Local\Keybase>keybase ctl stop --kill-kbfs
>- ERROR Timed out waiting for pipe '\\.\pipe\kbservice\Users\Deozaan\AppData\Local\Keybase\keybased.sock' to come available

Log ID: 1751ffa7cdf92c03a76fba1c

Temp files are still being created every few seconds. Or they were, until I tracked down the three files you mentioned in Task Manager and killed them manually.

Any other ideas?

**Comment by zanderz** [comment]
Afraid I'm running out of ideas. Totally baffled why keybase.exe cannot write to that directory; maybe some AV program is installed? If you really want to go deep you could install [Process Monitor](https://technet.microsoft.com/en-us/sysinternals/bb896645) to see exactly which process has that directory locked or opened exclusively.

**Comment by Deozaan** [comment]
But that's the weird thing. Keybase.exe **can** write to that directory. It does so frequently. Every ~10 seconds I get 5 new temp files written to that directory.

Could this have something to do with it using ReplaceIfExists, and since it doesn't exist, it doesn't replace it?

The only AV I have is Windows Defender. I've just tried disabling it with no visible difference in outcome.

But after removing everything and reinstalling several times--even using something like Revo Uninstaller to make sure nothing was left behind between reinstalls--there still is no `keybased.info` file, no `config.json` file, and no `updater.json` file. Additionally, I ran Process Monitor and there are other files in the `AppData/Local/Keybase` folder that Keybase tried to read or access that don't actually exist in the directory: `winmm.dll`, `WINMMBASE.dll`, `CRYPTSP.dll`, and `bcrypt.dll`.

And here's a weird anomaly I can't explain: It looks like Steam.exe tried to do something to keybase.exe in my `AppData/Local/Keybase` folder.

To be honest, I don't really know what I'm looking at in Process Monitor. As far as I can tell, all access in the `AppData/Local/Keybase` directory is working as it should, with the exception of trying to read files that don't exist there.

But maybe someone who understands these things better than I do will be able to glean more information from it than I can. Here are a couple of screenshots of some events (not just the highlighted ones) in Process Monitor which may or may not be useful:

![procmon - keybase files not found](https://cloud.githubusercontent.com/assets/227734/23636659/e42d8b48-0293-11e7-9bc7-10a8f3f71473.png)

![procmon - keybase temp files](https://cloud.githubusercontent.com/assets/227734/23636660/e6f335f8-0293-11e7-907d-3106892df82e.png)

**Comment by zanderz** [comment]
Thanks for those. If you google that error, it says `0xC0000368 is STATUS_MOUNT_POINT_NOT_RESOLVED: "The create operation failed because the name contained at least one mount point which resolves to a volume to which the specified device object is not attached."` 
Googling shows that many people report getting this when using cross-volume junctions. I think your filesystem may have problems.
https://forum.sysinternals.com/problems-with-junction_topic16205.htmlk

That is curious about Steam crawling all your directories. I don't think it is interfering, but it would creep me out.

**Comment by Deozaan** [comment]
Aha. See? I told you I didn't know what I was looking at. :)

I didn't realize that was an error. I thought if there was an error then the Result would say "Error" or "Failure" or something similar.

I'll see if I can fix my filesystem somehow. Thanks.

**Comment by Deozaan** [comment]
I recreated the Junction and now I can get past the Loading screen.

So the issue seems to be that the installer/updater broke/removed the Junction, then recreated the Keybase folder in its place, but didn't properly clean up the remnants of the Junction in the filesystem in the process.


But now it's asking me to login and seems to be generating new keys for my device, even though I copied my `.ss` and `.mpack` files back into the directory.

How do I restore my account with these files?

**Comment by zanderz** [comment]
@maxtaco @patrickxb are there other files or settings needed to restore a device?

**Comment by zanderz** [comment]
There are two shortcuts installed in `%APPDATA%\Microsoft\Windows\Start Menu\Programs\Startup` - one for the GUI and one for the background processes, named `KeybaseStartup.lnk`. It launches `%LOCALAPPDATA%\Keybase\runquiet.exe keybase.exe ctl watchdog2`. Runquiet is just a utility to cleanly background a console app in Windows. Again, lower-case keybase.exe does the important stuff; the GUI just drives it.

**Comment by Deozaan** [comment]
@zanderz I'm sorry, but I'm not making the connection of how that is related to this. I get the feeling that maybe your last comment was meant for another issue.

Or are you saying I need to add/modify those shortcuts to restore my account?

**Comment by zanderz** [comment]
Apologies @Deozaan , config.json needs some special entries. I have recreated it for you from publicly available info (your device ID, etc.) and can either post it here or send it some more private way if you prefer. Again, it is made of public, non-sensitive info.

**Comment by Deozaan** [comment]
I'd say to share it with me using Keybase, but I need my account restored first for that. :)

Go ahead and post it here, or maybe DM it to me on [Twitter](https://twitter.com/Deozaan) if you don't mind. It doesn't really matter to me since the information is non-sensitive.

**Comment by zanderz** [comment]
Seems to be too long to tweet. Try replacing config.json with this and restarting the service with `keybase ctl restart`:
```
{
    "current_user": "deozaan",
    "users": {
        "deozaan": {
            "device": "773e733db2b3320ac9837bfdce4b9f18",
            "id": "70dedcf125b25ccd87f0ca5cda2b1d19",
            "name": "deozaan",
            "salt": "6287ba5cda4f206558a879ca9075c657"
        }
    }
}
```

**Comment by Deozaan** [comment]
It worked! Thanks so much!

**Comment by ferrouswheel** [comment]
I am also in the same position. 75k files written to `~/AppData/Local/Keybase`. Can't get passed loading screen. I've tried uninstalling/installing several times, including deleting the detritus that is left over after uninstalling.

**Comment by maxtaco** [comment]
@ferrouswheel so sorry for this issue, it's quite embarrassing for us!

Are you also using Junctions?

Cc: @zanderz  and @gabriel (on the keybased.info write loop)

**Comment by zanderz** [comment]
@ferrouswheel Can you do `keybase log send` and provide the ID?

@Deozaan Just wondering if you moved `%LOCALAPPDATA%` to a junction after installing Keybase? As part of update, Keybase will remove the previous version's files and directories, so maybe that could have caused some problems.

**Comment by Deozaan** [comment]
@zanderz Windows won't let me actually make the %LOCALAPPDATA% directory itself a junction, so I have to make each subdirectory within it a junction. And since I can't know what folders will be created in there until after they've already been created by various programs, I have to wait until after things are installed and using the directory before moving it and creating the junctions.

Basically, any time I start running low on free space on my small SSD, the first place I look to free up room is in my %APPDATA% and %LOCALAPPDATA% folders and see what I can move and junction off.

I've been doing this for years and never had a problem with anything else.

**Comment by ferrouswheel** [comment]
@zanderz @maxtaco Cool I'll get the log and check if I'm using junctions this weekend (I only have the issue on my home pc).

I don't recall explicitly messing with junctions, but I did configure some of my home folders to use a different drive (as far as I know it wasn't `%LOCALAPPDATA%` being changed though).

**Comment by gabriel** [comment]
There is bug in safe write not cleaning up tmp file on rename error.. fix here: https://github.com/keybase/client/pull/6211

**Comment by ferrouswheel** [comment]
Using `dir /aL /s C:\` Windows indicates I am using junctions for both `C:\Users\USER\AppData\Roaming` and `C:\Users\USER\AppData\Local` among other locations.

The ID given by `keybase log send` is `6b494f45280f85816f61311c`.
